src/mapnik_image.cpp

```diff
@@ -1620,6 +1644,131 @@ void Image::EIO_AfterOpen(uv_work_t* req)
     delete closure;
 }
 
+// Read from a Buffer
+NAN_METHOD(Image::fromSVGBytesSync)
+{
+    NanScope();
+    NanReturnValue(_fromSVGSync(false, args));
+}
+
+// Read from a file
+NAN_METHOD(Image::fromSVGSync)
+{
+    NanScope();
+    NanReturnValue(_fromSVGSync(true, args));
+}
+
+Local<Value> Image::_fromSVGSync(bool fromFile, _NAN_METHOD_ARGS)
+{
+    NanEscapableScope();
+
+    if (!fromFile && (args.Length() < 1 || !args[0]->IsObject())) {
+        NanThrowTypeError("must provide a buffer argument");
+        return NanEscapeScope(NanUndefined());
+    }
+
+    if (fromFile && (args.Length() < 1 || !args[0]->IsString())) {
+        NanThrowTypeError("must provide a filename argument");
+        return NanEscapeScope(NanUndefined());
+    }
+
+
+    double scale = 1.0;
+    if (args.Length() >= 2) {
+        if (!args[1]->IsObject()) {
+            NanThrowTypeError("optional second arg must be an options object");
+            return NanEscapeScope(NanUndefined());
+        }
+        Local<Object> options = args[1]->ToObject();
+        if (options->Has(NanNew("scale")))
+        {
+            Local<Value> scale_opt = options->Get(NanNew("scale"));
+            if (!scale_opt->IsNumber()) {
+                NanThrowTypeError("'scale' must be an number");
+                return NanEscapeScope(NanUndefined());
+            }
+            scale = scale_opt->NumberValue();
+        }
+    }
+
+    try
+    {
+
+        //std::shared_ptr<mapnik::marker const> marker = mapnik::marker_cache::instance().find(svg_filename, false);
+
+        using namespace mapnik::svg;
+        mapnik::svg_path_ptr marker_path(std::make_shared<mapnik::svg_storage_type>());
+        vertex_stl_adapter<svg_path_storage> stl_storage(marker_path->source());
+        svg_path_adapter svg_path(stl_storage);
+        svg_converter_type svg(svg_path, marker_path->attributes());
+        svg_parser p(svg);
+        if (fromFile)
+        {
+            p.parse(TOSTR(args[0]));
```

Comment:
This seems like a possible situation for a segfault, tests should cover args[0] being a `null` object, or something else like a number as well.